### DISCUSSION TITLE
Unable to upload .ods

### DISCUSSION THREAD

**Discussion by nickyduff** [scientific-tool-set/scitos#39] [Q&A]
Hi there
I have Scitos v2.3.1 and I am unable to upload .ods file. I have been able to before, but now it's not working for me. Unsure what to do - any suggestions? 
Thank you
Nicky

**Comment by CarstenWickner** [comment]
> I have Scitos v2.3.1 and I am unable to upload .ods file. I have been able to before, but now it's not working for me.

Hi @nickyduff,
Can you please open a command prompt and start the application via
```
java -jar scitos.jar
```
and then try to upload again.
I'm assuming it fail fail again but at least provide some error output in the command prompt window.
Please share that error output then please.

Otherwise I've not much to go on here. 😬

**Reply by nickyduff** [reply]
Hi Carsten
Thank you for getting back to me. I can't seem to open it via the command prompt and copying in your text. It is sitting on my desktop and I can open it still, so unsure why the command prompt isn't working for me. 
I don't get an error message when I try to upload - nothing happens. 
I'm wondering whether I delete the programme and try downloading it again? 
Thank you
Nicky

**Reply by CarstenWickner** [reply]
Hi @nickyduff,

The given command assumes that the command prompt is in the context of the folder containing the `scitos.jar` file.
You might have to navigate there first via the `cd` command (followed by the path to that folder).

Downloading again is unlikely to help here I'm afraid, but you can of course try.

**Reply by nickyduff** [reply]
Hi @CarstenWickner 

Thank you for your reply. I managed to open SciTos via the command prompt, but no error message comes up when I try to upload with a .ods file. 

I found your sample .ods file you provided to someone who was having the same problem under the "issues" section on here. I can upload your sample.ods file, but when I try to add text to this sample.ods file, it doesn't work. 

When I save the .ods file, I get the attached message.

Thank you!

Nicky
![saving ods message](https://user-images.githubusercontent.com/52138141/180095272-6143e8c7-02f8-438f-80e4-c287611428c1.PNG)

**Reply by CarstenWickner** [reply]
Hi @nickyduff,

That warning from MS Excel is common and should usually not be an issue.
`.ods` like `.xlsx` is an XML-based file format. The mentioned "prolog" is at the top of the file and looks something like this (when you open it in a normal text editor, e.g., Notepad++):
```xml
<?xml version="1.0" encoding="UTF-8" standalone="yes" ?>
```
That's what the error message refers to:
> The selected File does not contain a valid SciTos Project. Content is not allowed in prolog.

It appears like the `.ods` files you're generating are invalid as there is something (maybe an invisible character) in front of the XML prolog/declaration. In Notepad++ (and other applications) you could make those (invisible) characters visible and manually remove them yourself.

Did you recently update your MS Excel?
Alternatively, you could download the free LibreOffice or OpenOffice distributions, which use `.ods` as their main file format for spreadsheets. Perhaps those are better suited to generate valid files.

**Reply by nickyduff** [reply]
Hi @CarstenWickner 

I downloaded Open Office and that works now, so it must be something with excel creating invalid .ods docs. 

Thank you for your help! As well as having SciTos as open access - we use it a lot in the lab to measure episodic detail. I've also been using it to count other things. 

Thank you

Nicky

**Reply by KristinaDMS** [reply]
HI Nicky and Carsten, I would like to say this solution worked for me. I downloaded Apache Open Office, opened the .xlsx file in there, and converted into an .ods. Then Scitos 2.4.4 opened a preview of my ODS file. Thanks!!
